lib/WiFiNative/src/WiFiClient.h

```diff
@@ -0,0 +1,145 @@
+    DESCRIPTION
+*******************************************************************************/
+/**
+ * @brief  Wifi Client for Native platform. Emulation of Arduino WiFiClient.
+ * @author Norbert Schulz <github@schulznorbert.de>
+ *
+ * @addtogroup HALSim
+ *
+ * @{
+ */
+#ifndef WIFI_CLIENT_H
+#define WIFI_CLIENT_H
+
+/******************************************************************************
+ * Compile Switches
+ *****************************************************************************/
+
+/******************************************************************************
+ * Includes
+ *****************************************************************************/
+
+#include <Arduino.h>
+#include <IPAddress.h>
+
+#include <poll.h>
+
+/******************************************************************************
+ * Macros
+ *****************************************************************************/
+
+/******************************************************************************
+ * Types and Classes
+ *****************************************************************************/
+
+/**
+ * WiFiClient client class compatible with Arduino. 
+ * 
+ * Only a subset of API functions is supported, based on the actual demand.
+ */
+class WiFiClient
+{
+    public:
+
+    /**
+     * Default constructor.
+     */
+    WiFiClient() :
+        m_poll {SOCK_INVAL, POLLIN, 0}
+    {
+    }
+
+    /**
+     * Default Destructor.
+     */
+    virtual ~WiFiClient()
+    {
+    }
+
+    /**
+     * Establish a connection.
+     * 
+     * @param[in] addr Port number.
+     * @param[in] port Port number.
+     * @return If successful returns 1, otherwise 0.
+     */
+    uint8_t connect(const IPAddress& addr, uint16_t port);
+
+    /**
+     * Indicate if client is connected.
+     * 
+     * @return If successful returns 1, otherwise 0.
+     */
+    uint8_t connected()
+    {
+        return (SOCK_INVAL != m_poll.fd) ? 1U : 0U;
+    }
+
+    /**
+     * Stop the WiF Client.
+     */
+    void stop();
+
+    /**
+     * Write bytes to stream.
+     * 
+     * @param[in] buffer Byte Array to send.
+     * @param[in] size Length of Buffer.
+     * @returns Number of bytes written
+     */
+    size_t write(const uint8_t* buffer, size_t size);
+
+
+    /**
+     * Check if there are available bytes in the Stream.
+     * 
+     * @returns Number of available bytes.
+     */
+    int available() const;
+
+    /**
+     * Read bytes from the incoming packet into a buffer.
+     *
+     * @param[out] buffer Buffer to read into.
+     * @param[in] size Length of the buffer.
+     *
+     * @return Number of bytes read or -1 on error.
+     */
+    int read(uint8_t* buffer, size_t size);
+
+private:
+    struct pollfd         m_poll;                       /**< The poll/socket structure. */
+
+    static const  int SOCK_INVAL = -1;                  /**< The invalid socket value   */
```

Comment:
One space too much between const and int.

Reply:
ok